### PR TITLE
[ACR-1595] Add animation encoding APIs for GIF and WEBP

### PR DESCRIPTION
**Pull request by jiyong2**
- ACR: ACR-1595

I missed new guide of encoding animation APIs.
So I updated the guide of encoding animation, please review this new guide.

### CODE REVIEW

File: docs/application/native/guides/multimedia/image-edit.md [hunk 1]
```diff
@@ -23,11 +23,12 @@ The main features of the Image Util API include:
...
+  You can [decode images](#decode), [encode them](#encode) and [encode animation](#animation) with the following formats:
 
   - Bitmap formats: YUV420, YUV422, RGB888, RGBA8888, BGRA8888, and ARGB8888
-  - Input image formats for decoding: JPEG, PNG, GIF, and BMP
-  - Output image formats for encoding: JPEG, PNG, GIF, BMP, and [animated GIF](#animation)
+  - Input image formats for decoding: JPEG, PNG, GIF, BMP and WEBP
+  - Output image formats for encoding: JPEG, PNG, GIF, BMP and WEBP
+  - Output image formats for encoding animation: [GIF and WEBP](#animation)
 
   > **Note**
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:33]: Global: Please ensure to use the latest note style all over the file. 

> [!NOTE]
>  your text ....

File: docs/application/native/guides/multimedia/image-edit.md [hunk 2]
```diff
@@ -471,7 +470,46 @@ To encode a raw image:
    ```
 
 <a name="animation"></a>
-## Encoding an Animated GIF
+## Encoding an Animated GIF or WEBP
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:473]: Global: Please use sentence case for all headings

File: docs/application/native/guides/multimedia/image-edit.md [hunk 3]
```diff
@@ -471,7 +470,46 @@ To encode a raw image:
...
+
+To encode an animated GIF or WEBP image:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:475]:
```suggestion 
To encode an animated GIF or WEBP image, follow these steps:

File: docs/application/native/guides/multimedia/image-edit.md [hunk 4]
```diff
@@ -471,7 +470,46 @@ To encode a raw image:
...
+
+1. Create an encoding handle using `image_util_anim_encode_create()`:
+
+   ```
+   image_util_anim_encode_h anim_encode_h = NULL;
+   ret = image_util_anim_encode_create(IMAGE_UTIL_ANIM_WEBP, &anim_encode_h);
+   ```
+
+2. Additionally, you can set the loop count, background color or lossless `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:484]: Can we write like this?
```suggestion
2.  Set the loop count, background color or lossless `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:

File: docs/application/native/guides/multimedia/image-edit.md [hunk 5]
```diff
@@ -471,7 +470,46 @@ To encode a raw image:
...
+
+   ```
+   ret = image_util_anim_encode_set_loop_count(anim_encode_h, 10);
+   ```
+
+   > [!NOTE]
+   > The background color and lossless are supported for the WEBP animation.
+   > The default of the loop count is 0(infinite) and the default of lossless is false(lossy). The default of background color is no use.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:492]:
```suggestion
   > [!NOTE]
   > The background color and lossless compression is supported for the WEBP animation.
   > The default loop count is `0` (infinite), and the default of lossless compression is `false` (lossy). The default background color is `no use`.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@jiyong2 Request you to incorporate the comments.

File: docs/application/native/guides/multimedia/image-edit.md [hunk 6]
```diff
@@ -23,11 +23,12 @@ The main features of the Image Util API include:
 
 - Decoding from a file or memory and encoding to a file or memory
 
-  You can [decode images](#decode) and [encode them](#encode) with the following formats:
+  You can [decode images](#decode), [encode them](#encode) and [encode animation](#animation) with the following formats:
```
[the remark this answers is not in the document]
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: Thank you for your comment, I updated it.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 1 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: Thank you for your comment, I updated the latest note style all over the file.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 2 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md:469]: ok! thank you for the comment.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 3 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: ok! thank you for the comment.
I updated it.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 4 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: I updated it.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 5 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: Thank you for the comment. I did it.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
I have checked the changed parts, all seems well but please change all headings to sentence case.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
Request you to incorporate the comments, also please see the sentence case comment.

**Comment by nawab-samsung** [comment]
@jiyong2 Gentle reminder:  Please incorporate the comments.

File: docs/application/native/guides/multimedia/image-edit.md [hunk 7]
```diff
@@ -23,14 +23,14 @@ The main features of the Image Util API include:
...
 - Decoding from a file or memory and encoding to a file or memory
 
-  You can [decode images](#decode) and [encode them](#encode) with the following formats:
+  You can [decode images](#decode), [encode them](#encode), and [encode animation](#animation) with the following formats:
 
   - Bitmap formats: YUV420, YUV422, RGB888, RGBA8888, BGRA8888, and ARGB8888
-  - Input image formats for decoding: JPEG, PNG, GIF, and BMP
-  - Output image formats for encoding: JPEG, PNG, GIF, BMP, and [animated GIF](#animation)
+  - Input image formats for decoding: JPEG, PNG, GIF, BMP and WEBP
+  - Output image formats for encoding: JPEG, PNG, GIF, BMP and WEBP
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:30]:
```suggestion
  - Input image formats for decoding: JPEG, PNG, GIF, BMP, and WEBP
  - Output image formats for encoding: JPEG, PNG, GIF, BMP, and WEBP

File: docs/application/native/guides/multimedia/image-edit.md [hunk 8]
```diff
@@ -23,14 +23,14 @@ The main features of the Image Util API include:
...
+  - Output image formats for encoding animation: [GIF and WEBP](#animation)
 
-  > **Note**
-  >
+  > [!NOTE]
   > Pay attention to how the [image quality depends on the size](#quality) and compression ratio.
```
[the remark this answers is not in the document]
Comment (nawab-samsung) [reply docs/application/native/guides/multimedia/image-edit.md:34]: @jiyong2 Please incorporate the missed comment.

File: docs/application/native/guides/multimedia/image-edit.md [hunk 9]
```diff
@@ -471,7 +466,47 @@ To encode a raw image:
...
+To encode an animated GIF or WEBP image, follow these steps:
+
+1. Create an encoding handle using `image_util_anim_encode_create()`:
+
+   ```
+   image_util_anim_encode_h anim_encode_h = NULL;
+   ret = image_util_anim_encode_create(IMAGE_UTIL_ANIM_WEBP, &anim_encode_h);
+   ```
+
+2. Set the loop count, background color or lossless `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:480]: 2. Set the loop count, background color or lossless `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:

Can we say 
I might be wrong, but request you to check and modify if needed 
```suggestion
2. Set the loop count, the background color or the lossless compression properties with `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:

File: docs/application/native/guides/multimedia/image-edit.md [hunk 10]
```diff
@@ -471,7 +466,47 @@ To encode a raw image:
...
+
+   ```
+   ret = image_util_anim_encode_set_loop_count(anim_encode_h, 10);
+   ```
+
+   > [!NOTE]
+   > The background color and lossless compression is supported for the WEBP animation.
+   > The default of the loop count is '0' (infinite), and the default of lossless compression is 'false' (lossy).
+   > The default background color is 'no use'.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:489]:
```suggestion
   > [!NOTE]
   > The background color and the lossless compression is supported for the WEBP animation.
   > The default of the loop count is `0`(infinite), and the default of lossless compression is `false` (lossy).
   > The default background color is `no use`.

File: docs/application/native/guides/multimedia/image-edit.md [hunk 11]
```diff
@@ -471,7 +466,47 @@ To encode a raw image:
...
+   ret = image_util_anim_encode_save_to_file(anim_encode_h, path);
+   ```
+
+5. After the encoding is complete, destroy the encoding handle using `image_util_anim_encode_destroy()`:
+
+   ```
+   ret = image_util_anim_encode_destroy(anim_encode_h);
+   ```
+
+## Encoding an Animated GIF simply
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:509]: ## Encoding an Animated GIF

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@jiyong2 Please incorporate the comments, a few were missed last time.

**Comment by nawab-samsung** [comment]
@jiyong2 Request you to incorporate the comments.

**Comment by nawab-samsung** [comment]
Gentle Reminder: @jiyong2 Request you to incorporate the comments.

**Comment by nawab-samsung** [comment]
@jiyong2 Request you to incorporate the comments.

**Comment by nawab-samsung** [comment]
@jiyong2 Request you to incorporate the comments.

**Comment by nawab-samsung** [comment]
@jiyong2 Gentle Reminder: Request you to incorporate the comments.

**Comment by annie-samsung** [comment]
@jiyong2, could you please close this PR or fix the comments?

File: docs/application/native/guides/multimedia/image-edit.md (hunk 10 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: OK, 'the' article has been added, others were modified.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: ok, the missed comment has been added.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 7 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: ok, the comma has been added.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 11 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: 'simply' was removed.

**Comment by jiyong2** [comment]
@jiyong2 Gentle Reminder: Request you to incorporate the comments.
Sorry for the late reply. I modified the comments.

**Comment by nawab-samsung** [comment]
@jiyong2 thanks for incorporating the comments, I will review the changes and get back to you if anything more is needed. 

Thanks

File: docs/application/native/guides/multimedia/image-edit.md [hunk 12]
```diff
@@ -471,6 +466,46 @@ To encode a raw image:
...
+2. Set the loop count, the background color or the lossless compression properties with `image_util_anim_encode_set_loop_count()`, `image_util_anim_encode_set_background_color()` or `image_util_anim_encode_set_lossless()`:
+
+   ```
+   ret = image_util_anim_encode_set_loop_count(anim_encode_h, 10);
+   ```
+
+   > [!NOTE]
+   > The background color and the lossless compression is supported for the WEBP animation.
+   > The default of the loop count is `0` (infinite), and the default of lossless compression is 'false' (lossy).
+   > The default background color is `no use`.
```
Comment (nawab-samsung) [inline docs/application/native/guides/multimedia/image-edit.md:489]:
```suggestion
   > [!NOTE]
   > The background color and the lossless compression is supported for the WEBP animation.
   > The default of the loop count is `0` (infinite), and the default of lossless compression is `false` (lossy).
   > The default background color is `no use`.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@jiyong2 Please incorporate the comment, rest looks good to me.

File: docs/application/native/guides/multimedia/image-edit.md (hunk 12 above)
Comment (jiyong2) [reply docs/application/native/guides/multimedia/image-edit.md]: ok, I will fix it.

**Comment by jiyong2** [comment]
@nawab-samsung, Thank you! I fixed all the comments. Please check and review.

**Comment by nawab-samsung** [comment]
@jiyong2 why have you raised this PR for 6.0 branch? Shouldn't it be 6.5 ?

**Comment by nawab-samsung** [comment]
@jiyong2 why have you raised this PR for 6.0 branch? Shouldn't it be 6.5 ?

**Comment by jiyong2** [comment]
> @jiyong2 why have you raised this PR for 6.0 branch? Any specific reason for the same ? Shouldn't it be 6.5 ?

This change takes effect since Tizen 6.0. So I raised this PR for 6.0 branch.

**Comment by nawab-samsung** [comment]
> > @jiyong2 why have you raised this PR for 6.0 branch? Any specific reason for the same ? Shouldn't it be 6.5 ?
> 
> This change takes effect since Tizen 6.0. So I raised this PR for 6.0 branch.

@annie-samsung Could you please help us here, As I know we are currently having 6.5 as latest version. You can give us a solution here.

**Comment by annie-samsung** [comment]
@jiyong2 could you please raise this PR for master branch. 6.0_prepare was used when 6.0 was not yet public on the Tizen Website, last October 6.0 was made public so all the contents of that branch were merged to master and 6.0_prepare is no longer used.

**Comment by jiyong2** [comment]
> @jiyong2 could you please raise this PR for master branch. 6.0_prepare was used when 6.0 was not yet public on the Tizen Website, last October 6.0 was made public so all the contents of that branch were merged to master and 6.0_prepare is no longer used.

ok I will raise this PR for master branch and share it.

**Comment by jiyong2** [comment]
I update this PR to #1478 for master branch.
So I will close this PR.

**Comment by annie-samsung** [comment]
> I update this PR to #1478 for master branch.
> So I will close this PR.
@jiyong2, yes please! Kindly close this PR. I have approved and merged PR #1478.

**Comment by jiyong2** [comment]
#1478  has been merged. so I close this PR.

**Pull request closed without merging**